cmd/signature-v4-utils.go

```diff
@@ -260,3 +262,46 @@ func signV4TrimAll(input string) string {
 	// unicode.IsSpace() internally here) to one space and return
 	return strings.Join(strings.Fields(input), " ")
 }
+
+// checkMetaHeaders will check if the metadata from header/url is equal with the one from signed headers
+func checkMetaHeaders(signedHeadersMap http.Header, r *http.Request) APIErrorCode {
+	metaHeaders := make(http.Header)
```

Comment:
```suggestion
	metaHeaders := make(http.Header, len(signedHeadersMap))
```

Though in general I would prefer if you didn't allocate this at all. Since you aren't checking for values that are in the signature, but not present in the header, it shouldn't be needed.

This should be the equivalent AFAICT, but without the alloc
```
	// take values from http header
	for k, got := range r.Header {
		if stringsHasPrefixFold(k, "X-Amz-Meta-") {
			if signedHeadersMap.Get(k) == got[0] { // Maybe fail if len(got) != 1?
				continue
			}
			return ErrUnsignedHeaders
		}
	}
	//... check URL values in a similar way.
```